### ISSUE TITLE
Import renaming causes false negatives with `min_max`

### ISSUE THREAD

**Issue by apasel422** [Manishearth/rust-clippy#446]
For example, it doesn't trigger with:

```rust
use std::cmp::max;
use std::cmp::min as my_min;

fn main() {
   let x = 2;
   my_min(1, max(3, x));
}
```
This may apply to other lints as well.

**Comment by llogiq** [comment]
Yeah, basically everywhere we just match the path without looking up the actual Method.

**Comment by apasel422** [comment]
What is the proper way to check this sort of thing?

**Comment by Manishearth** [comment]
Use `match_def_path` on the exprcall
